conans/client/build/cmake_flags.py

```diff
@@ -204,7 +204,8 @@ def _cmake_cross_build_defines(self, cmake_version):
         else:  # detect if we are cross building and the system name and version
             skip_x64_x86 = os_ in ['Windows', 'Linux']
             if cross_building(self._conanfile, skip_x64_x86=skip_x64_x86):  # We are cross building
-                apple_system_name = "Darwin" if Version(cmake_version) < Version("3.14") else None
+                apple_system_name = "Darwin" if cmake_version and Version(cmake_version) < Version(
+                    "3.14") or not cmake_version else None
```

Comment:
```suggestion
                    "3.14") else None
```

Reply:
I think that If I change that it's going to resolve as if the cmake version is > than 3.14 and I think we want to have the current behaviour when the cmake version call fails and we don't have it. I have added one more checks to the tests:

```python
definitions = definitions_builder.get_definitions(None)
self.assertEqual("Darwin", definitions["CMAKE_SYSTEM_NAME"])
```